bitcode/CMakeLists.txt

```diff
@@ -109,18 +105,15 @@ set(OCML_LIB_PATHS
   ${CMAKE_BINARY_DIR}/bitcode/ROCm-Device-Libs/oclc/oclc_isa_version_803.lib.bc
 )
 
-if(CHIP_ENABLE_NON_COMPLIANT_DEVICELIB_CODE)
-    # add support for calling certain LLVM builtins that are not supported by HIP/CUDA
-    add_custom_command(
-    OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/BC/c_to_opencl.bc"
-    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/c_to_opencl.c"
-    COMMAND "${CMAKE_CXX_COMPILER}" ${BITCODE_C_COMPILE_FLAGS}
-    -o "${CMAKE_CURRENT_BINARY_DIR}/BC/c_to_opencl.bc"
-    -c "${CMAKE_CURRENT_SOURCE_DIR}/c_to_opencl.c"
-    COMMENT "Building c_to_opencl.bc"
-    VERBATIM)
-    list(APPEND DEPEND_LIST "${CMAKE_CURRENT_BINARY_DIR}/BC/c_to_opencl.bc")
-endif()
+add_custom_command(
```

Comment:
Why are we getting rid of the option? These are still non-compliant, right?

Reply:
> Why are we getting rid of the option?

These are required for using compiler built-ins (see [c_to_opencl.def](https://github.com/CHIP-SPV/chipStar/blob/0dba21d5c36121b9f0323acb2319e2bbcb00fc83/bitcode/c_to_opencl.def#L23-L39) for explanation).

> These are still non-compliant, right?

Non-compliant respect to what?

Reply:
So why not keep the option? In atomics.hh we still rely on this option:

```
#ifdef CHIP_ENABLE_NON_COMPLIANT_DEVICELIB_CODE
// At least rocPRIM tests call the unsigned long variant although it's not
// listed in the user manual. Annoyingly, size_t is typically defined as
// unsigned long.
// FIXME: We should check that unsigned long is 64bits for the host.
extern "C++" inline __device__ unsigned long atomicAdd(unsigned long *address,
                                                       unsigned long val) {
  return __chip_atomic_add_l(address, val);
}
#endif
```


> Non-compliant respect to what?

https://github.com/CHIP-SPV/chipStar/issues/700

"when CHIP_ENABLE_NON_COMPLIANT_DEVICELIB_CODE` was defined, which brought in the atomicAdd() for longs (see https://github.com/CHIP-SPV/chipStar/commit/71b5c82fd0235c12d0e20f2adeffa75eaa8eef12) which is not defined in the HIP spec."

Reply:
If the concern here is that removing the option in the bitcode library would expose non-compliant functions to the users, that shouldn’t be happening because they are not declared in the HIP headers. Therefore, removing the CHIP_ENABLE_NON_COMPLIANT_DEVICELIB_CODE option check shouldn't have negative consequences, AFAICS.